### DISCUSSION TITLE
v2.4.0-Beta 1 needs work...

### DISCUSSION THREAD

**Discussion by rsaid99** [OCJvanDijk/Brightness-Sync#37] [General]
I've had to stop using this beta on my M1 MBA.  While it works for a while, it's unstable to the point of not being worth running.  Its been hard to pinpoint exactly what is going on, but here are some symptoms I'm seeing:

* The Brightness Offset slider will eventually go away (along with the app functionality).  Sometimes selecting Reset will bring it back, but often times it won't.
* The app eventually (within 24 hours) will just stop working.
* There is odd behavior around plugging/unplugging the USBC cable as well, but its not consistent enough to describe.

Hopefully these can be resolved over time.  I'm happy to try some different test versions if there is still limited access to an M1 Mac.

**Comment by OCJvanDijk** [comment]
Sounds like issues around display detection. M1 Macs seem to have a new system for (fast) switching displays, but I changed nothing in that department. I only fixed the actual brightness get/set. I've been running the beta all this time on an Intel Mac with no problems.

When it stops working, does quitting and restarting the app solve the problem?

**Reply by rsaid99** [reply]
Yes, quitting and restarting works 100%

**Comment by rsaid99** [comment]
After approximately 20 days of not using Brightness Sync, I wanted to report that there seems to be some issues with either my setup, Big Sur or the M1.  At least a couple times of week I see one of these issues (without Brightness Sync):
* Adjusting the brightness via bluetooth keyboard doesn't register with my MBA
*  Adjusting the brightness via the bluetooth keyboard brings up the brightness display on the Ultrafine, and increments/decrements, but doesn't actually change the brightness.
* My external monitor isn't recognized at all.

All of these require an unplug of the USBC, and re-plug.

Based on these observations I'm pretty sure that whatever is going on was probably making life for Brightness Sync worse.  Hopefully Big Sur 11.3 will clean some of this up.

**Comment by Maqsim** [comment]
Hey @OCJvanDijk. Seems like Launch at login doesn't work (M1). Will verify it once more and let you know

**Reply by rsaid99** [reply]
After every MacOS update I give BrightnessSync a try again to see if anything changes.  I'm happy to report that over time, its behavior is getting more stable and usable.

While this may seem strange given that the beta hasn't changed, I should mention that in my experience Apple is still fixing problems with how MacOS handles plugging/unplugging the Ultrafine (2nd gen in my case), and even maintaining a functional brightness control connection between the M1 MBA and the Ultrafine.  Even without BrightnessSync running I often lose the ability to control the monitor brightness via the keyboard.

A year ago, these problems made it impossible for BrightnessSync to stay functional for long.  However as of today it stays running and rarely have to use reset or re-boot it.  The most common issue I see now is that BrightnessSync loses my offset setting, and sets it to something else.  Minor all things considered, and perhaps even fixable on the app side.

All that to say, I hope functionality continues to improve as Apple fixes things, and that more test versions can be put out to try to corral the last remaining issues.
